tests/integration_tests/functional/test_rng.py

```diff
@@ -122,3 +123,111 @@ def test_rng_snapshot(test_microvm_with_rng, microvm_factory):
+    """
+    if units == "kB/s":
+        return 1000
+
+    if units == "MB/s":
+        return 1000 * 1000
+
+    if units == "GB/s":
+        return 1000 * 1000 * 1000
+
+    raise Exception("Unknown units")
+
+
+def _process_dd_output(out):
+    """
+    Parse the output of `dd` and return the achieved throughput in
+    KB/sec
+    """
+    report = out.splitlines()[-1].split(" ")
+    (value, units) = (report[-2], report[-1])
+
+    return float(value) * _throughput_units_multiplier(units) / 1000
+
+
+def _get_throughput(ssh, random_bytes):
+    """
+    Request `random_bytes` from `/dev/hwrng` and return the achieved
+    throughput in KB/sec
+    """
+    cmd = "dd if=/dev/hwrng of=/dev/null bs={} count=1".format(random_bytes)
+    exit_code, _, stderr = ssh.execute_command(cmd)
+    assert exit_code == 0, stderr.read()
+
+    # dd gives its output on stderr
+    return _process_dd_output(stderr.read())
+
+
+def _check_entropy_rate_limited(ssh, random_bytes, expected_kbps):
+    """
+    Ask for `random_bytes` from `/dev/hwrng` in the guest and check
+    that achieved throughput is within a 10% of the expected throughput.
+
+    NOTE: 10% is an arbitrarily selected limit which should be safe enough,
+    so that we don't run into many intermittent CI failures.
+    """
+    measured_kbps = _get_throughput(ssh, random_bytes)
+    assert (
+        _get_percentage_difference(measured_kbps, expected_kbps) <= 10
+    ), "Expected {} KB/s, measured {} KB/s".format(expected_kbps, measured_kbps)
+
+
+def _rate_limiter_id(rate_limiter):
+    """
+    Helper function to return a name for the rate_limiter to be
+    used as an id for parametrized tests.
+    """
+    size = rate_limiter["bandwidth"]["size"]
+    refill_time = rate_limiter["bandwidth"]["refill_time"]
+
+    return "{} KB/sec".format(size / (refill_time / 1000.0) / 1000.0)
+
+
+@pytest.mark.skipif(
+    INSTANCE_TYPE == "c7g.metal" and HOST_KERNEL == "4.14",
+    reason="c7g requires no SVE 5.10 kernel",
+)
+@pytest.mark.parametrize(
+    "rate_limiter",
+    [
+        {"bandwidth": {"size": 1000, "refill_time": 100}},
+        {"bandwidth": {"size": 10000, "refill_time": 100}},
+        {"bandwidth": {"size": 100000, "refill_time": 100}},
+    ],
+    ids=_rate_limiter_id,
+)
+def test_rng_bw_rate_limiter(test_microvm_with_rng, rate_limiter):
+    """
+    Test that rate limiter without initial burst budget works
+
+    @type: functional
+    """
+    vm = test_microvm_with_rng
+    _start_vm_with_rng(vm, rate_limiter)
+
+    size = rate_limiter["bandwidth"]["size"]
+    expected_kbps = 10 * size / 1000.0
+    # Run the test asking twice the set limit, so that we make
+    # sure we exercise the rate limiter. Otherwise, i.e. if we ask
+    # less bytes than the size of the token bucket we will not ever
+    # get rate limited and achieve the maximum possible throughput
+    _check_entropy_rate_limited(vm.ssh, 2 * size * 10, expected_kbps)
```

Comment:
Maybe add a comment about where the 10 comes from (I think its 1000/refill_time, in which case shouldn't be actually state it as that? Right now, if the parameterization of the test changes, it will break - also, same algebraic simplifcation as above is possible then :))

Reply:
I re-worked all of this, because it was sloppy. I now just pass `size` and execute `dd` with that size and `count = 100`. My assumption is that running any request 100 times should average out any outliers of the rate limiter. WDYT?

Reply:
You are right, I was just fixing in my head the value of `refill_time` in `100 msec`. Will update it.